OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -185,6 +185,23 @@ internal static void UpgradeActorRules(ModData modData, int engineVersion, ref L
 					}
 				}
 
+				// GivesExperience was changed from having a FriendlyFire bool to a Stances array.
+				if (engineVersion < 20160702 && node.Key.StartsWith("GivesExperience"))
+				{
+					var ff = "FriendlyFire";
+					var ffNode = node.Value.Nodes.FirstOrDefault(n => n.Key == ff);
+					if (ffNode != null)
+					{
+						if (FieldLoader.GetValue<bool>(ff, ffNode.Value.Value))
```

Comment:
Wouldn't the actor we inherit from (who had `true` before) get `Neutral, Enemy, Ally`?
And we inherit that then (although we only want `false`/`Neutral, Enemy`).